flate/inflate.go

```diff
@@ -175,10 +179,10 @@ func (h *huffmanDecoder) init(lengths []int) bool {
 	}
 
 	h.maxRead = min
-	chunks := h.chunks[:]
-	for i := range chunks {
-		chunks[i] = 0
-	}
+
+	// instead of iterating over the whole array, just copy already null-filled
+	// slice in it.
+	copy(h.chunks[:], zeroChunks)
```

Comment:
Copying is pointless and just slower. This loop is already being picked up by the compiler.

If you want a neat oneliner use `*h.chunks = [huffmanNumChunks]uint16{}`. It produces the exact same code as the current (using DUFFZERO).

Reply:
I did some benchmarks, and original approach is the most performant (0.3ns faster). Copying and zeroing via for-loop without slicing are at the same level of 8.5ns, and reassigning is ~2x slower (14ns). Sorry for ignorance, I'm gonna return everything back just as it was:)